### PR TITLE
Serial port manual configurable

### PR DESCRIPTION
**Pull request by setola**
Hello,
I've added this feature to your project cause my arduino was on /dev/ttyACM0 and the library was not able to detect it.

Thank you for your work, it's awesome!
Keep it going!

### CODE REVIEW

**Comment by nmec** [comment]
Thanks for posting this, got things working for me in Ubuntu. Hope this can be merged soon.

**Comment by flosse** [comment]
+1

File: README.md [hunk 1]
```diff
@@ -36,7 +36,14 @@ The way this works is simple (in theory, not in practice). The Arduino listens f
...
 ##board
 
-Right now, the board library will attempt to autodiscover the Arduino. I'm going to make it configurable, don't worry.
+````javascript
+var board = new arduino.Board({
+  device: "ACM"
+});
+````
+The 'device' wich arduino is attached on. 
+Here you can set the exact device (ex. `ttyACM0` in my case) or the initial part of it (ex `ttyACM`)
```
Comment (ecto) [inline README.md]: So in the example, this will connect to `/dev/ACM`? I think this could be a little more clear.

**Comment by ecto** [comment]
Hey guys, I don't have my Arduino right now (it's halfway across the country). Can you confirm that this patch is legit? Also @setola, could you clarify the README section I commented on? I'll push this to npm afterward!

**Comment by ecto** [comment]
Also it appears there is a merge conflict after merging https://github.com/ecto/duino/pull/26 :persevere:

File: README.md (hunk 1 above)
Comment (setola) [inline README.md]: No, in the example the library searches for Arduino in every device that has 'ACM' string in its name.

A better documentation maybe:

The board library will attempt to autodiscover the Arduino.
Here you can set a regex filter to limit the library scan only matching devices.
It uses grep syntax. 
If this parameter is empty the board library will attempt to autodiscover the Arduino by quering every device containing 'usb' in its name.

**Comment by setola** [comment]
I think that the conflict on #26 is because i forked the project before the #26 was committed.

**Comment by setola** [comment]
Conflict should be solved on [99dc318](https://github.com/setola/duino/commit/99dc31805d92053aab5aa1e198f32dc3c7a63386) commit, I've also improved documentation of 'device' parameter.
Hope notthing is missing...

cheers

**Comment by bigmonkeyboy** [comment]
This works really well... +1 to merge.

**Comment by anpross** [comment]
oh well - should have checked the pull requests first - i just build my own patch to fix the same problem...

**Comment by poul-kg** [comment]
Why fixes by @setola  are still not merged? Is this project dead?

**Comment by setola** [comment]
try to ping @ecto to see if he's not so busy as usually :) :)
:+1:

**Comment by ecto** [comment]
Merged and pushed to npm @ 0.0.9. Thanks @setola!

**Comment by setola** [comment]
again a big thank for your work @ecto 
grats for the new version!!!

**Pull request merged**